ambry-store/src/main/java/com.github.ambry.store/IndexSegment.java

```diff
@@ -253,6 +264,19 @@ long getLastModifiedTime() {
   }
 
   /**
+   * The resetKey for the index segment.
+   * @return the reset key for the index segment which is a {@link Pair} of StoreKey and
+   * {@link PersistentIndex.IndexEntryType}
+   */
+  Pair<StoreKey, PersistentIndex.IndexEntryType> getResetKey() {
+    if (version == (short) 1 && resetKey != null) {
+      return new Pair<>(resetKey.getFirst(), PersistentIndex.IndexEntryType.values()[resetKey.getSecond()]);
```

Comment:
Sorry. can you clarify what is your suggestion here. Store a pair of (StoreKey, IndexEntry) for resetKey and during persistence, just persist the StoreKey and type? Then this code will differ depending on whether it is active segment or persisted one. Hence, I had to choose this way.